File: sav_frontend/src/types/report.ts
export type Report = {
  id: string;
  interventionId?: string;
  clientId?: string;
  technicianId?: string;
  isWarranty?: boolean;
  total?: number;
  generatedAt?: string;
  url?: string;
  title?: string;
  clientNom?: string;
  clientEmail?: string;
  clientTelephone?: string;
};

export type ReportRequest = {
  interventionId?: string;
  clientId?: string;
  isWarranty?: boolean;
  total?: number;
  technicianId?: string;
  title?: string;
};

File: Service_apres_vente_back/ReportingAPI/Models/Report.cs
namespace ReportingAPI.Models
{
    public class Report
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid InterventionId { get; set; }
        public Guid ClientId { get; set; }
        // Optional: link report to a technician (for technicien-specific reports)
        public Guid? TechnicianId { get; set; }
        public bool IsWarranty { get; set; }
        public decimal Total { get; set; }
        public DateTime GeneratedAt { get; set; } = DateTime.UtcNow;
        public string Url { get; set; } = string.Empty;
        public string? Title { get; set; }
    }
}
